pkg/controller/volume/expand/expand_controller.go

```diff
@@ -171,12 +171,6 @@ func (expc *expandController) deletePVC(obj interface{}) {
 }
 
 func (expc *expandController) pvcUpdate(oldObj, newObj interface{}) {
-	oldPvc, ok := oldObj.(*v1.PersistentVolumeClaim)
-
-	if oldPvc == nil || !ok {
-		return
-	}
-
```

Comment:
I expected the old and new objects to be compared to see if they needed to be added to the resize map. @gnufied

Reply:
It does gets check before being added to the map - https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/expand/cache/volume_resize_map.go#L96

Reply:
Okay, we don't compare old and new pvc objects before adding to resize map and that is intentional. The reason is - we can't only rely on events for processing resize requests and there is a different loop that also polls for all PVCs in cluster and checks if pvc's requested size and actual capacity don't match. If PVC is in that state, it gets added to resize map.

This is as per design and there are checks in place to make sure everything works as expected.